src/main.cpp

```diff
@@ -1843,54 +2031,78 @@ void static InvalidBlockFound(CBlockIndex *pindex, const CValidationState &state
     }
 }
 
-void UpdateCoins(const CTransaction& tx, CValidationState &state, CCoinsViewCache &inputs, CTxUndo &txundo, int nHeight)
+void UpdateCoins(const CTransactionBase& txBase, CValidationState &state, CCoinsViewCache &inputs, CTxUndo &txundo, int nHeight)
 {
     // mark inputs spent
-    if (!tx.IsCoinBase()) {
-        txundo.vprevout.reserve(tx.vin.size());
-        BOOST_FOREACH(const CTxIn &txin, tx.vin) {
+    if (!txBase.IsCoinBase())
+    {
+        txundo.vprevout.reserve(txBase.GetVin().size());
+        for(const CTxIn &txin: txBase.GetVin())
+        {
             CCoinsModifier coins = inputs.ModifyCoins(txin.prevout.hash);
             unsigned nPos = txin.prevout.n;
+            assert(coins->IsAvailable(nPos));
 
-            if (nPos >= coins->vout.size() || coins->vout[nPos].IsNull())
-                assert(false);
             // mark an outpoint spent, and construct undo information
             txundo.vprevout.push_back(CTxInUndo(coins->vout[nPos]));
+            LogPrint("cert", "%s():%d - spending inputs from [%s]\n", __func__, __LINE__, txin.prevout.hash.ToString());
             coins->Spend(nPos);
-            if (coins->vout.size() == 0) {
+            if (coins->vout.size() == 0 || coins->vout[nPos].isFromBackwardTransfer) {
```

Comment:
don't test anymore isFromBwt but copy in the undo the backwardTrasferStartIdx field taken from pruned coins